### ISSUE TITLE
how to remove bounding box, label in SAHI

### ISSUE THREAD

**Issue by dimasadef72** [ultralytics/ultralytics#22195]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to ask, how do I save the results without displaying the bounding box and label? I want to save only the segmentation mask and remove the bounding box and label.

this is my code : 
`from sahi.predict import get_sliced_prediction
from sahi import AutoDetectionModel
import os
import torch
from utils import read_geotiff_metadata, save_as_geotiff

Create main result folder
os.makedirs("result", exist_ok=True)

 Create incremental subfolder inside result
counter = 0
result_dir = f"result/run{counter}"
while os.path.exists(result_dir):
    counter += 1
    result_dir = f"result/run{counter}"
os.makedirs(result_dir, exist_ok=True)
print(f"Results will be saved to: {result_dir}")

Check GPU availability
print(f"CUDA available: {torch.cuda.is_available()}")
if torch.cuda.is_available():
    print(f"GPU device: {torch.cuda.get_device_name(0)}")
    print(f"GPU memory: {torch.cuda.get_device_properties(0).total_memory / 1024**3:.1f} GB")
else:
    print("GPU not available, using CPU")

Setup SAHI AutoDetectionModel
detection_model = AutoDetectionModel.from_pretrained(
    model_type="ultralytics",
    model_path="models/best (3).pt",  # your trained model
    confidence_threshold=0.3,
    device="cuda:0",  # using GPU
)

# Input source - large image for tiled inference
source = "test.png"  # your large area image

print(f"Starting SAHI inference on {source}...")

 Perform sliced prediction with SAHI
result = get_sliced_prediction(
    source,
    detection_model,
    slice_height=640,
    slice_width=640,
    overlap_height_ratio=0.2,
    overlap_width_ratio=0.2,
)

 Export visualization results with mask only (no labels)
result.export_visuals(
    export_dir=f"{result_dir}/"
)
print("SAHI inference completed!")

Print detection statistics
print(f"Total detections: {len(result.object_prediction_list)}")
for i, prediction in enumerate(result.object_prediction_list):
    print(f"Detection {i+1}: Class={prediction.category.name}, Confidence={prediction.score.value:.3f}")

print(f"Results saved in '{result_dir}/' directory")`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dimasadef72, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly. An Ultralytics engineer will also follow up here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a custom inference/visualization ❓ Question (e.g., saving only masks), please share a bit more context to help us assist:
-Exact `ultralytics` version (`python -c "import ultralytics, sys; print(ultralytics.__version__)"`) and PyTorch/CUDA versions
-Whether you’re using SAHI (version) and any SAHI-specific visualization params you’ve tried
-A minimal snippet that reproduces the current output and a short description/screenshot of expected vs. actual results
-A sample image (or a small crop) and your current visualization artifacts

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by dimasadef72** [comment]
<img width="325" height="157" alt="Image" src="https://github.com/user-attachments/assets/fbe82954-ee89-43a2-ac83-ce1ddc24c205" />

i want to remove bounding box and label, that this example

**Comment by dhaan-ish** [comment]
If you want to remove bounding box and label then that is the original image, right? You can use the original image if you dont want bounding box and lable.

**Comment by dimasadef72** [comment]
I want to remove the bounding box and the label, but I want to keep only the mask segmentation

**Comment by dhaan-ish** [comment]
```python
class PredictionResult:
    def export_visuals(
        self,
        export_dir: str,
        text_size: float | None = None,
        rect_th: int | None = None,
        hide_labels: bool = False,
        hide_conf: bool = False,
        file_name: str = "prediction_visual",
    ):
        """
        Args:
            export_dir: directory for resulting visualization to be exported
            text_size: size of the category name over box
            rect_th: rectangle thickness
            hide_labels: hide labels
            hide_conf: hide confidence
            file_name: saving name
        Returns:
            None
        """
```
This is the method of export_visuals. You can rewrite the code as follow

```python
result.export_visuals(
export_dir=f"{result_dir}/"
rect_th=0, #0 thickness to remove the bounding box
hide_labels=True,
hide_conf=True
)
```

You can refer at https://github.com/obss/sahi/blob/main/sahi/prediction.py

**Comment by glenn-jocher** [comment]
Thanks both — to render masks only with SAHI, call `export_visuals` with zero box thickness and hidden labels/conf (there’s a comma missing after `export_dir` in the snippet above):
```python
# masks-only overlay (no boxes / no labels)
result.export_visuals(
    export_dir=result_dir,   # <-- comma
    rect_th=0,
    hide_labels=True,
    hide_conf=True,
    file_name="prediction_visual"
)
```
If you need mask images themselves (no background), save each instance mask from `object_prediction_list`:
```python
from PIL import Image
import os, numpy as np

os.makedirs(f"{result_dir}/masks", exist_ok=True)
for i, p in enumerate(result.object_prediction_list):
    if getattr(p, "mask", None) is None:
        continue
    mask = (p.mask.bool_mask.astype(np.uint8) * 255)  # binary mask
    Image.fromarray(mask).save(f"{result_dir}/masks/mask_{i:05d}.png")
```
For a quick refresher on YOLO+SAHI usage see the guide [Using YOLO11 with SAHI for Sliced Inference]. For SAHI’s mask field on predictions, see examples using `prediction.mask.bool_mask`. ([docs.ultralytics.com](https://docs.ultralytics.com/guides/sahi-tiled-inference/))

If you’re not tied to SAHI’s visualizer, YOLO’s own renderer can also produce masks-only overlays via `Results.plot(boxes=False, labels=False, masks=True)`; see the [Results.plot() arguments] page.
